hal/src/rtl872x/gpio_hal.cpp

```diff
@@ -141,18 +141,19 @@ int hal_gpio_configure(hal_pin_t pin, const hal_gpio_config_t* conf, void* reser
         GPIO_InitStruct.GPIO_Pin = rtlPin;
 
         switch (mode) {
-            case OUTPUT:
-            case OUTPUT_OPEN_DRAIN: {
+            case OUTPUT: {
                 GPIO_InitStruct.GPIO_Mode = GPIO_Mode_OUT;
                 GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_NOPULL;
                 break;
             }
-            case INPUT: {
+            case INPUT:
+            case OUTPUT_OPEN_DRAIN: {
                 GPIO_InitStruct.GPIO_Mode = GPIO_Mode_IN;
                 GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_NOPULL;
                 break;
             }
-            case INPUT_PULLUP: {
+            case INPUT_PULLUP:
+            case OUTPUT_OPEN_DRAIN_PULLUP: {

```

Comment:
Let's also add a quick test to `wiring/no_fixture/gpio.cpp` to configure to `OUTPUT_OPEN_DRAIN_PULLUP` and validate whether `digitalRead()`/`digitalWrite()` work as intended.

Reply:
If there is a pin with an external pull-up, might want to also test `OUTPUT_OPEN_DRAIN`, but that's probably only SDA/SCL on platforms with PMIC/FG, so probably can be omitted.